src/end_entity.rs

```diff
@@ -74,6 +75,37 @@ impl<'a> EndEntityCert<'a> {
         &self.inner
     }
 
+    /// Verifies that the end-entity certificate is valid for use against the
+    /// specified EKU (Extended Key Use).
```

Comment:
small nit: we should match the RFC language and use "usage" instead of "use". I also think it might be more natural to write the full version and then the acronym rather than the other way around.

```suggestion
    /// specified extended key usage (EKU).
```